streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -811,8 +811,28 @@ private void prepareChangelogs(final Set<ChangelogMetadata> newPartitionsToResto
         }
     }
 
+    void maybeTriggerOnRestoreEnd(final Collection<TopicPartition> changelogPartitions) {
```

Comment:
Alright, the situation here is that we need to make sure we toggle bulk loading off for any active restoring tasks that convert to standby. Rather than try and force a direct call to `toggleForBulkLoading` on the store itself I figured we should just call `onRestoreEnd`. Technically, restoration _is_ ending. It just happens to be due to type transition, rather than restore completion.
I figured this might be relevant for users of custom stores, which might do something similar to bulk loading that they wish to turn off for standbys. But since this is only relevant to the underlying store, and doesn't mean we have actually finished restoring a task, we should only call the specific store's listener -- and _not_ the user registered global listener.
WDYT?